### ISSUE TITLE
SketchMapper broken with macOS Catalina

### ISSUE THREAD

**Issue by MattyB95** [josephtaylor/sketch-mapper#28]
I am currently unable to run SketchMapperTest while operating on macOS Catalina error below is produced when tried. I am not able to reproduce this error on Windows!

> 2020-02-10 18:14:33.656 java[78362:2117301] WARNING: NSWindow drag regions should only be invalidated on the Main Thread! This will throw an exception in the future. Called from (
	0   AppKit                              0x00007fff391f0265 -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 371
	1   AppKit                              0x00007fff391d7bc5 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1416
	2   AppKit                              0x00007fff391d7637 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
	3   libnativewindow_macosx.jnilib       0x000000010f37c3fe Java_jogamp_nativewindow_macosx_OSXUtil_CreateNSWindow0 + 398
	4   ???                                 0x000000010f4eb667 0x0 + 4551784039
)

**Comment by MattyB95** [comment]
Looking in more detail seems to be related to the below and looks like it is a broader Processing issue; however, it currently means I am unable to run SketchMapper on my Mac :(

> WARNING: An illegal reflective access operation has occurred
> WARNING:` Illegal reflective access by processing.opengl.PSurfaceJOGL (file:/Applications/Processing.app/Contents/Java/core.jar) to method sun.java2d.opengl.CGLGraphicsConfig.getContextCapabilities()
> WARNING: Please consider reporting this to the maintainers of processing.opengl.PSurfaceJOGL
> WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> WARNING: All illegal access operations will be denied in a future release

**Comment by josephtaylor** [comment]
Those all look like warnings though, is there any other output in the terminal when it crashes?

**Comment by josephtaylor** [comment]
I have a mac but I can't put Catalina on it cause it'll break Ableton and Max :smile:

**Comment by josephtaylor** [comment]
It looks like it might be related to [this bug in jogl](https://jogamp.org/bugzilla/show_bug.cgi?id=1393) which they'll have fixed in their next release version `2.4.0`, but I can't tell when that's gonna happen. Their last release was in 2015.

**Comment by MattyB95** [comment]
Yes, I believe you are correct and thanks for your advice. There is no other warnings in the console and Processing locks up and goes into "Not Responding" mode.

I will close this because it is related to Processing in general and not SketchMapper. 

I was able to get some workaround by using the new alpha version of Processing (4.0 alpha 1). However, it still would lockup in the processing IDE. Yet, by importing the libraries and using an alternative Java IDE (IntelliJ), I was able to get SketchMapper working again <- May be useful if anybody else experiences this issue.
